KS-1-5 convert seconds to hours and minutes

CasTeTRU opened the pull request.

CasTeTRU committed:
KS-1-5 convert seconds to hours and minutes

ievetrovdev commented on src/main/kotlin/lesson_1/lesson1_task5.kt:
```diff
@@ -0,0 +1,12 @@
+    val secondsInSpace = 6489
+    val hoursInSpace = secondsInSpace / 3600        //целые часы
+    val minutesLeft = secondsInSpace % 3600 / 60    //остаток минут
+    val secondsLeft = secondsInSpace % 60           //остаток секунд
```
Все окей, но чтобы снизить риск ошибок, предлагаю еще вынести 3600 и 60 в константы. Кроме того это [магические числа](https://ru.wikipedia.org/wiki/%D0%9C%D0%B0%D0%B3%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%BE%D0%B5_%D1%87%D0%B8%D1%81%D0%BB%D0%BE_(%D0%BF%D1%80%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5)#%D0%9F%D0%BB%D0%BE%D1%85%D0%B0%D1%8F_%D0%BF%D1%80%D0%B0%D0%BA%D1%82%D0%B8%D0%BA%D0%B0_%D0%BF%D1%80%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D0%BC%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D1%8F), что тоже не является хорошей практикой. А также комментарии эти излишние, они говорят просто явные вещи, их нужно убрать.

ievetrovdev commented on src/main/kotlin/lesson_1/lesson1_task5.kt:
```diff
@@ -0,0 +1,12 @@
+    val minutesLeft = secondsInSpace % 3600 / 60    //остаток минут
+    val secondsLeft = secondsInSpace % 60           //остаток секунд
+
+//Отображаем часы,минуты,секунды двумя цифрами
```
Нужно убрать комментарий

ievetrovdev commented on src/main/kotlin/lesson_1/lesson1_task5.kt:
```diff
@@ -0,0 +1,12 @@
+    val secondsLeft = secondsInSpace % 60           //остаток секунд
+
+//Отображаем часы,минуты,секунды двумя цифрами
+   println("${hoursInSpace.toString().padStart(2, '0')}:${minutesLeft.toString().padStart(2, '0')}:${secondsLeft.toString().padStart(2, '0')}")
```
Так как это 5 задача ("со звездочкой") не по уровню урока, то предлагаю использовать более элегантное решение). (может потребовать дополнительного ресерча). Для корректного вывода нуля вначале попробуй посмотреть в сторону String.format().

ievetrovdev commented on src/main/kotlin/lesson_1/lesson1_task5.kt:
```diff
@@ -0,0 +1,12 @@
+
+fun main() {
+
+    val secondsInSpace = 6489
```
Точно 6489, а не 6480?

ievetrovdev reviewed (changes requested):
Я оставил комментарии для доработки. чтобы правильно залить изменения, надо:
 - переключиться в Идее на нужную ветку
 - внести изменения и сделать коммит в эту ветку (следить чтобы в стейдж не добавились лишние задачи, которые не относятся к данной задаче)
 - отправить ссылку в бот на этот же пулл реквест к соответствующей задаче

CasTeTRU committed:
KS-1-5-Time spent in space
Changed output of seconds, minutes, hours using String.format() function
Added two constants to get rid of magic numbers

ievetrovdev commented on src/main/kotlin/lesson_1/lesson1_task5.kt:
```diff
@@ -0,0 +1,13 @@
+    val hoursInSpace = String.format("%02d", secondsInSpace / secondsInHour)
+    val minutesLeft = String.format("%02d", secondsInSpace % secondsInHour / secondsInMinute)
+    val secondsLeft = String.format("%02d", secondsInSpace % secondsInMinute)
+    println("$hoursInSpace:$minutesLeft:$secondsLeft")
```
Форматирование использовано верно, но так как это задачка повышенной сложности, доведем вывод до более оптимального состояния.
- можно отформатировать в одну строку
- можно сразу использовать интерполяцию при выводе
Вот так:
println("%02d:%02d:%02d".format(hours, minutes, seconds))

ievetrovdev commented on src/main/kotlin/lesson_1/lesson1_task5.kt:
```diff
@@ -0,0 +1,13 @@
+package org.example.lesson_1
+
+import java.text.DecimalFormat
```
Если все сделать верно, то не понадобится импорт

CasTeTRU committed:
KS-1-5-convert-seconds-to-hours-and-minutes
Переделан вывод так что .format теперь применяется к строке вывода, а не к каждой переменной по отдельности

ievetrovdev reviewed (approved):
По задаче все отлично, аппрув!

Теперь эту ветку можно мержить в мастер. напомню, перед созданием новой ветки надо переключиться в мастер, обновить его (чтобы слитые изменения прилетели в локальный проект) и только тогда бранчиться
